test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/TypeValidatorTests.cs

```diff
@@ -49,267 +41,236 @@ public void TestInit()
+        Verify(0 == _warnings.Count);
     }
 
-    [TestMethod]
     public void WhenInternalDiscoveryIsEnabledIsValidTestClassShouldReturnTrueForNestedInternalTestClasses()
     {
         var typeValidator = new TypeValidator(_mockReflectHelper.Object, true);
 
         SetupTestClass();
-        Assert.IsTrue(typeValidator.IsValidTestClass(typeof(OuterClass.NestedInternalClass), _warnings));
+        Verify(typeValidator.IsValidTestClass(typeof(OuterClass.NestedInternalClass), _warnings));
     }
 
-    [TestMethod]
     public void WhenInternalDiscoveryIsEnabledIsValidTestClassShouldReturnFalseForPrivateTestClasses()
     {
         var typeValidator = new TypeValidator(_mockReflectHelper.Object, true);
 
         var nestedPrivateClassType = Assembly.GetExecutingAssembly().GetTypes().First(t => t.Name == "NestedPrivateClass");
 
         SetupTestClass();
-        Assert.IsFalse(typeValidator.IsValidTestClass(nestedPrivateClassType, _warnings));
+        Verify(!typeValidator.IsValidTestClass(nestedPrivateClassType, _warnings));
     }
 
-    [TestMethod]
     public void WhenInternalDiscoveryIsEnabledIsValidTestClassShouldReturnFalseForInaccessibleTestClasses()
     {
         var typeValidator = new TypeValidator(_mockReflectHelper.Object, true);
 
         var inaccessibleClassType = Assembly.GetExecutingAssembly().GetTypes().First(t => t.Name == "InaccessiblePublicClass");
 
         SetupTestClass();
-        Assert.IsFalse(typeValidator.IsValidTestClass(inaccessibleClassType, _warnings));
+        Verify(!typeValidator.IsValidTestClass(inaccessibleClassType, _warnings));
     }
 
-    [TestMethod]
     public void WhenInternalDiscoveryIsEnabledIsValidTestClassShouldNotReportWarningsForNestedInternalTestClasses()
     {
         var typeValidator = new TypeValidator(_mockReflectHelper.Object, true);
 
         SetupTestClass();
         typeValidator.IsValidTestClass(typeof(OuterClass.NestedInternalClass), _warnings);
-        Assert.AreEqual(0, _warnings.Count);
+        Verify(0 == _warnings.Count);
     }
 
     #endregion
 
     #region Generic types
 
-    [TestMethod]
     public void IsValidTestClassShouldReturnFalseForNonAbstractGenericTypes()
     {
         SetupTestClass();
-        Assert.IsFalse(_typeValidator.IsValidTestClass(typeof(GenericClass<>), _warnings));
+        Verify(!_typeValidator.IsValidTestClass(typeof(GenericClass<>), _warnings));
     }
 
-    [TestMethod]
     public void IsValidTestClassShouldReportWarningsForNonAbstractGenericTypes()
     {
         SetupTestClass();
         _typeValidator.IsValidTestClass(typeof(GenericClass<>), _warnings);
-        Assert.AreEqual(1, _warnings.Count);
-        CollectionAssert.Contains(_warnings, string.Format(Resource.UTA_ErrorNonPublicTestClass, typeof(GenericClass<>).FullName));
+        Verify(1 == _warnings.Count);
+        Verify(_warnings.Contains(string.Format(Resource.UTA_ErrorNonPublicTestClass, typeof(GenericClass<>).FullName)));
```

Comment:
See comment about collections equality.

Reply:
this isn't a collections equality, it's a contains function.
so, no changing needed, right?